add GOVERNANCE.md

haircommander opened the pull request:
<!--  Thanks for sending a pull request!

Please be aware that we're following the Kubernetes guidelines of contributing
to this project. This means that we have to use this mandatory template for all
of our pull requests.

Please also make sure you've read and understood our contributing guidelines
(https://github.com/cri-o/cri-o/blob/master/CONTRIBUTING.md) as well as ensuring
that all your commits are signed with `git commit -s`.

Here are some additional tips for you:

- If this is your first time, please read our contributor guidelines:
  https://git.k8s.io/community/contributors/guide#your-first-contribution and
  developer guide
  https://git.k8s.io/community/contributors/devel/development.md#development-guide
- Please label this pull request according to what type of issue you are
  addressing, especially if this is a release targeted pull request. For
  reference on required PR/issue labels, read here:
  https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
- If you want *faster* PR reviews, read how:
  https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
- If the PR is unfinished, see how to mark it:
  https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

<!--
Uncomment only one `/kind <>` line, hit enter to put that in a new line, and
remove leading whitespace from that line:
-->

> /kind api-change
> /kind bug
> /kind cleanup
> /kind dependency-change
> /kind deprecation
> /kind design

/kind documentation

> /kind failing-test
> /kind feature
> /kind flake

#### What this PR does / why we need it:
adds GOVERNANCE.md
#### Which issue(s) this PR fixes:

<!--
Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->

<!--
Fixes #
or
None
-->

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes see:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
none
```

haircommander commented:
```
ASK [clone kubernetes source repo] ********************************************
task path: /go/src/github.com/cri-o/cri-o/contrib/test/integration/build/kubernetes.yml:3
fatal: [localhost]: FAILED! => {"changed": false, "cmd": "/usr/bin/git ls-remote origin -h refs/heads/master", "msg": "fatal: unable to access 'https://github.com/kubernetes/kubernetes.git/': transfer closed with outstanding read data remaining", "rc": 128, "stderr": "fatal: unable to access 'https://github.com/kubernetes/kubernetes.git/': transfer closed with outstanding read data remaining\n", "stderr_lines": ["fatal: unable to access 'https://github.com/kubernetes/kubernetes.git/': transfer closed with outstanding read data remaining"], "stdout": "", "stdout_lines": []}
```
:cry: 
/retest

haircommander commented:
/retest

fidencio commented:
As someone who has been around but not spending as much as I want in the project, I simply love this initiative!

fidencio commented on GOVERNANCE.md:
```diff
@@ -37,7 +37,19 @@
+
+The CRI-O community has the emeritus maintainer tier to recognize the impact and input early maintainers have had on the project, while also being realistic about their current committment level.
+
+There is no formal process for becoming an emeritus maintainer: a maintainer simply slows down on their contributions but does not remove themselves from the [OWNERS file](OWNERS.md).
```
@haircommander, I really like the "Emeritus" idea, GNOME dos something quite similar.

If I may make a suggestion, I'd prefer having the Emeritus Maintainers listed as so. By doing this new contributors will know who they can ping, who they can ask for help, etc.  With no clear distinction between both may be a slightly confusing for new contributors (I must admit it was slightly confusing for me when I first arrived here a few months ago).

TomSweeneyRedHat commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,68 @@
+* During GitHub issue triage, apply all applicable [labels](https://github.com/cri-o/cri-o/labels) to each new issue. Labels are extremely useful for future issue follow up. Which labels to apply is somewhat subjective so just use your best judgment. A few of the most important labels that are not self explanatory are:
+  * **good first issue**: Mark any issue that can reasonably be accomplished by a new contributor with this label.
+  * **help wanted**: Unless it is immediately obvious that someone is going to work on an issue (and if so assign it), mark it help wanted.
+  * **CRI change**: If sufficiently fixing the issue involves a change to the CRI, this label is given. Often, maintainers won't have time to make such a change, but the change could be seen as good and as such the issue isn't closed.
```
Do you have a 'kind/feature' label and do you want to also highlight that?

haircommander replied:
that's more for PRs, for issues we have `area/vm` but I don't think it's common enough to warrant adding right now

TomSweeneyRedHat commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,68 @@
+
+# Conflict resolution and voting
+
+In general, we prefer that technical issues and maintainer membership are amicably worked out between the persons involved. If a dispute cannot be decided independently, the maintainers can be called in to decide an issue. If the maintainers themselves cannot decide an issue, the issue will be resolved by voting. The voting process is a simple majority in which each maintainer receives one vote.  ...
```
Any tie breaker procedures?  If you list them, you'll never need them, if you don't....
Perhaps just "In the case of a tie vote, the issue will be considered to have not passed."

saschagrunert commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,68 @@
+# CRI-O contributor tiers
```
Let's link the file from the README.md with a short sentence describing its intention.

TomSweeneyRedHat replied:
Good thought @saschagrunert I'd also suggest a link and a bit of explanatory verbiage on CONTRIBUTING.md

haircommander commented:
a year and a quarter later, I've addressed the comments, and updated with some feedback from today's community meeting :smiley: 
PTAL @cri-o/cri-o-maintainers

saschagrunert commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,99 @@
+# CRI-O GOVERNANCE.md
+
+CRI-O, being a project whose purpose is linked to fulfilling the needs of Kubernetes, has a governance structure that mirrors many SIGs and sub-projects of Kubernetes. This document highlights important aspects of, and serves as a summary of the [Kubernetes' community documentation](https://www.kubernetes.dev/docs.
```
```suggestion
CRI-O, being a project whose purpose is linked to fulfilling the needs of Kubernetes, has a governance structure that mirrors many SIGs and sub-projects of Kubernetes. This document highlights important aspects of, and serves as a summary of the [Kubernetes' community documentation](https://www.kubernetes.dev/docs).
```

haircommander replied:
good catch!

saschagrunert commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,99 @@
+
+* Once you are a reviewer, the CRI-O community will expect you to start contributing PRs, taking issues and doing code reviews under the guidance of current approvers.
+
+* Monitor our channel (#crio) on the Kubernetes Slack instance and answer questions where possible (delayed response is perfectly acceptable).
```
Let's add a link to the slack channel

saschagrunert commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,99 @@
+
+### Approver responsibilities
+
+* All of the responsibilities of the reviewers, and...
```
```suggestion
Approvers have all of the responsibilities of the reviewers, as well as:
```

saschagrunert reviewed (commented):
Awesome, just a few nits otherwise LGTM

fgiudici commented on .github/ISSUE_TEMPLATE/membership.yml:
```diff
@@ -0,0 +1,42 @@
+      required: true
+    - label: I have one sponsor from the list of approvers in at least one OWNERS file within one of the CRI-O repository
+      required: true
+    - label: I have spoken to my sponsor ahead of this application, and they have agreed to sponsor my application
```
Nit:
```suggestion
    - label: I have spoken to my sponsor ahead of this application, and he has agreed to sponsor my application
```

haircommander replied:
please my friend, allow me to introduce you to the [singular they](https://en.wikipedia.org/wiki/Singular_they), a gender neutral alternative to gendered pronouns. It's become the idiomatic way to have documentation be non-gendered and more inclusive :slightly_smiling_face:

fgiudici replied:
ups, shame upon me 😅!

haircommander replied:
no shame, only a teachable moment :smile:

fgiudici commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,99 @@
+### How to become a reviewer
+
+* If you'd like to have a more active role in the community, and one day hope to become an approver, you may become a reviewer to build trust and deepen your knowledge of the code base. 
+* To become a reviewer, an you may file an issue with [this template](https://github.com/cri-o/cri-o/issues/new?assignees=&labels=area%2Fgithub-membership&template=membership.yml&title=REQUEST%3A+New+membership+for+%3Cyour-GH-handle%3E).
```
```suggestion
* To become a reviewer, you may file an issue with [this template](https://github.com/cri-o/cri-o/issues/new?assignees=&labels=area%2Fgithub-membership&template=membership.yml&title=REQUEST%3A+New+membership+for+%3Cyour-GH-handle%3E).
```

haircommander commented:
updated!

fgiudici reviewed (commented):
LGTM, great job!

fidencio commented:
@haircommander, the document is in a very good shape.  One thing that I am still missing, though, is information on how to find the list of reviewers / approvers.  And, yes, I know,  such a list doesn't exist right now.

So, on this, where do you plan to have such a list?  I am thinking mostly on the scenario where I have interest in one area and I want to find folks from the community with knowledge in that area to give me some guidance.

haircommander commented:
ah I see, I was planning on having https://github.com/cri-o/cri-o/blob/main/OWNERS be the authoritative document

haircommander commented:
updated!

fidencio commented:
LGTM, thanks @haircommander!

fidencio commented:
@haircommander, one more thing, actually, the document doesn't cover the transition between reviewer -> contributor, maintainer -> reviewer or maintainer -> emeritus maintainer.

I think that would be nice to have documented as it will help to evaluate the situation of all the current members prsent in the https://github.com/cri-o/cri-o/blob/main/OWNERS file.

haircommander commented:
> 

don't the sections "how to become x" cover that?

fidencio commented:
> don't the sections "how to become x" cover that?

Kind of.  I'm thinking on the discussion we had Yesterday on the meeting.
Let's say, "12 months with zero contributions" revokes your status as approver / reviewer.  approvers are automatically moved to the emeritus status, for instance; reviewers get back to the contributor status.

Do you think it'd make sense to mention / make that explicit?

haircommander commented:
okay, I've added a demotion section.

fidencio commented:
Thanks, @haircommander!

I think it covers everything I could think of.

saschagrunert reviewed (approved):
/lgtm
